CHANGELOG.md

```diff
@@ -0,0 +1,40 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
+and follows a [Backwards Compatability Policy]
+(https://docs.solana.com/developing/backwards-compatibility)
+
+## Adding to this Changelog
+### Audience
+* Entries in this log are intended to be easily understood by consensus
+validators, rpc operators, and dapp developers.
+
+### Noteworthy
+* Changes that relate to a SIMD, or add a feature gate should add an entry to
+this log. Other changes that are relevant to the aforementioned audiences should
+add an entry at the discretion of authors and reviewers.
+
+### Instructions
+* Update this log in the same pull request that implements the change. If the
+change is spread over several pull requests update this log in the one that
+makes the feature code complete.
+* Add notes to the [Unreleased] section in each branch that you merge to.
+  * Add a description of your change to the Changes section
+  * If your change is likely to require operators to update their configs add to
+  the Upgrade Notes
+
+## How To Maintain This Changelog
+* When a release is tagged and the version is bumped:
+  * Date and tag the [Unreleased] section with the new release including a link
+  to the release page
+  * Create new [Unreleased] section
+change
+## Unreleased
+* Changes
+  * Added a changelog!
+* Upgrade Notes
+
+## Released
+### 2023-05-31 - [1.16.0](https://github.com/solana-labs/solana/releases/tag/v1.16.0)
```

Comment:
Over time the release section would only contain the x.y.0 release right? We don't add a new entry for patch releases? How about DRYing this out and just linking generically to https://github.com/solana-labs/solana/releases

Reply:
> All notable changes to this project will be documented in this file.

I think this should be our guiding principle. If we have the changelog in each branch, and use it to document "all notable changes" it reduces mental load on both authors and consumers. The only question of where to find information is which branch to look at, but anyone reading the changelog likely already knows our branching stradegy.

I'm envisioning:
* The log maintains a full history since its inception (newest content at the top)
* Both bug fixes and new features are included (if they're notable)
* Patch releases get a new section (maybe skip the new section if there's no new content?)
* Henceforth the x.y.0 release should be the mergebase of master and vx.y

So the master version of CHANGELOG.md will include only the x.y.0 releases, but then the release branches will each contain a few more entries for bug fixes that were backported. That way the changelog mimics the code changes and the branching strategy is familiar and consistent.

Reply:
Ah, ok. Got it and I think I like it. One matter though: we bump crate versions *after* tagging a release, but with changelog would be introducing a step *before* tagging a release to update the changelog with the new release. We're 100% going to screw this up, a lot. 

Instead how about we bump the changelog at the same time as the crate versions? There's never an "Unreleased" section at all then, the most recent section (like the latest crate version of a branch) is the next anticipated release version